source/codegen/templates/service_helpers.mako

```diff
@@ -485,6 +488,22 @@ ${initialize_standard_input_param(function_name, parameter)}
 % endif
 </%def>
 
+## Initialize an input parameter with the 'two-dimension' size mechanism.
+<%def name="initialize_two_dimension_input_param(function_name, parameter, parameters)">\
+<%
+  parameter_name = common_helpers.get_cpp_local_name(parameter)
+  size_param_name = common_helpers.get_size_param(parameter)
+  size_param = common_helpers.get_param_with_name(parameters, size_param_name)
+  size_field_name = common_helpers.get_grpc_field_name(size_param)
+%>\
+${initialize_standard_input_param(function_name, parameter)}
+      auto total_length = std::accumulate(request->${size_field_name}().cbegin(), request->${size_field_name}().cend(), 0);
+
+      if (total_length != request->${parameter_name}_size()) {
+        return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The total size of ${parameter_name} did not match the exected size from ${size_field_name}");
```

Comment:
I'm trying to think of a better message here. Imagine a user that is as confused about this API as we were trying to figure it out.

"did not match the expected sizes" -> Just saying "sizes" instead of "size" helps a little bit.

I think either "sum" or "2D" should be in here somewhere.

> "The total size of the two-dimensional array ${parameter_name} does not match the expected size from the sum of ${size_field_name}"

Something like that.

Reply:
I like alex's recommendation

Reply:
I've updated this description so as to be more descriptive and helpful to the user.